ASTInfoGenerator/DeclParser/DeclParser.py

```diff
@@ -0,0 +1,382 @@
+from DeclParser.DeclType import DeclType
+from ASTInfoGenerator.Debug import *
+from ASTInfoGenerator.Specifiers.SpecifierManager import *
+
+from ASTInfoGenerator.DeclParser.DeclTypes.FunctionDecl import *
+from ASTInfoGenerator.DeclParser.DeclTypes.RecordDecl import *
+from ASTInfoGenerator.DeclParser.DeclTypes.ParmVarDecl import *
+from ASTInfoGenerator.DeclParser.DeclTypes.TypedefDecl import *
+from ASTInfoGenerator.DeclParser.DeclTypes.FieldDecl import *
+from ASTInfoGenerator.DeclParser.DeclTypes.StructVarDecl import *
+from ASTInfoGenerator.DeclParser.DeclTypes.AnnotateAttr import *
+
+
+class DeclParser:
+    def __init__(self):
+        self.__param_index = None
+        self.__field_index = None
+
+    def _parse_function_decl(self, function_decl: str) -> Specifier:
+        # Assumption: functions that are extern are not relevant to our analysis
+        if function_decl.endswith("extern"):
+            return None
```

Comment:
Why? `extern` doesn't really mean anything on functions - isn't external linkage the default?

Reply:
My understanding of `extern` might be straight out of 1999 though

Reply:
I cannot recall why I did this in the past and removing those LoC did not change much output, so I'll be removing it.

Reply:
Nevermind, I have found this definitely makes a difference. Way more functions are included in the final output, but I have no reason to not include it because it has an "extern." Many of these functions with "extern" are from standard C include files that do not concern our analysis, but it's very well possible a user writes a function with extern that we _should_ consider

Reply:
We can create a list but we don't know if that function came from what the user defined or if it's from the standard library. The AST does not preserve `#include` tags

Reply:
Is it really a problem if we include functions from the standard library in the output? I agree they're clutter, but it's not _wrong_, correct?

Reply:
Correct, it's not wrong to. But in the future I'll consider looking into seeing if I could filter them out, considering it changes AST info output from ~100 lines to 1000 lines